blocks/logic.ts

```diff
@@ -550,31 +566,36 @@ const CONTROLS_IF_TOOLTIP_EXTENSION = function() {
 
 Extensions.register('controls_if_tooltip', CONTROLS_IF_TOOLTIP_EXTENSION);
 
+/** Type of a block that has LOGIC_COMPARE_ONCHANGE_MIXIN */
+type CompareBlock = Block&CompareMixin;
+interface CompareMixin extends CompareMixinType {
+  prevBlocks_: Array<Block|null>;
```

Comment:
I think this should also be `|undefined`

Reply:
Agreed—but again, marked optional via `prevBlocks_?:` rather than `|undefined`, for clarity.